src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -178,6 +190,56 @@ public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String d
         }
     }
 
+    public DRSAccessURL getAccessUrlForObjectId(AuthenticatedUserRequest authUser, String objectId, String accessId) {
+        DRSObject object = lookupObjectByDrsId(authUser, objectId, false);
+
+        DrsId drsId = drsIdService.fromObjectId(objectId);
+        Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
+        SnapshotProject snapshotProject = snapshotService.retrieveAvailableSnapshotProject(snapshot.getId());
+        GoogleProjectResource projectResource = snapshot.getProjectResource();
+
+        List<DRSAccessMethod> matchingAccessMethods = getAccessMethodsMatchingAccessId(accessId, object);
+        if (matchingAccessMethods.size() > 0) {
+            Storage storage = StorageOptions.newBuilder().setProjectId(projectResource.getGoogleProjectId()).build().getService();
+            String urlString = matchingAccessMethods.get(0).getAccessUrl().toString();
+            Pattern gsBucketRegex = Pattern.compile("(?<=//)(.*?)(?=/)");
+            Matcher bucketMatcher =  gsBucketRegex.matcher(urlString);
+
+            if (bucketMatcher.find()) {
+                String gsBucket = bucketMatcher.group();
+                Pattern gsPathRegex = Pattern.compile(String.format("(?<=%s/)(.*)", gsBucket));
+                Matcher pathMatcher =  gsPathRegex.matcher(urlString);
+
+                if (pathMatcher.find()) {
+                    String path = pathMatcher.group();
+                    // Define resource
+                    BlobInfo blobInfo = BlobInfo.newBuilder(BlobId.of(gsBucket, path)).build();
+
+                    URL url =
+                        storage.signUrl(blobInfo, 15, TimeUnit.MINUTES, Storage.SignUrlOption.withV4Signature());
+
+                    return new DRSAccessURL().url(url.toString());
+                }
+                else {
+                    throw new IllegalArgumentException("Was not able to find bucket");
+                }
+            } else {
+                throw new IllegalArgumentException("Was not able to find bucket");
+            }
+        } else {
+            throw new IllegalArgumentException("No matching access ID " + accessId + " was found on object " + objectId);
+        }
+    }
+
+    private List<DRSAccessMethod> getAccessMethodsMatchingAccessId(String accessId,
+        DRSObject object) {
+        return object.getAccessMethods().stream()
+            .filter(
+                drsAccessMethod -> drsAccessMethod.getType().equals(TypeEnum.GS) && drsAccessMethod
```

Comment:
looking ahead, we don't want to filter on GS since this won't work for Azure access methods